PennMobile/New Group/PAC Code/PacCodeNetworkManager.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  PacCodeNetworkManager.swift
+//  PennMobile
+//
+//  Created by CHOI Jongmin on 1/3/2020.
+//  Copyright © 2020 PennLabs. All rights reserved.
+//
+
+import SwiftSoup
+
+class PacCodeNetworkManager {
+    static let instance = PacCodeNetworkManager()
+    private init() {}
+}
+
+extension PacCodeNetworkManager: PennAuthRequestable {
+
+    private var pacURL: String {
+        return "https://penncard.apps.upenn.edu/penncard/jsp/fast2.do?fastStart=pacExpress"
+    }
+    
+    private var shibbolethUrl: String {
+        return "https://penncard.apps.upenn.edu/penncard/jsp/fast2.do/Shibboleth.sso/SAML2/POST"
+    }
+    
+    func getPacCode(callback: @escaping ((_ code: String?) -> Void)) {
+        makeAuthRequest(targetUrl: pacURL, shibbolethUrl: shibbolethUrl) { (data, response, error) in
+            if let httpResponse = response as? HTTPURLResponse {
+                if httpResponse.statusCode == 200 {
+                    if let data = data, let html = NSString(data: data, encoding: String.Encoding.utf8.rawValue) as String? {
+                        do {
+                            let pacCode = try self.findPacCode(from: html)
+                            return callback(pacCode)
```

Comment:
We want to start using Result types in the app instead of optional values. Here is a [link](https://www.hackingwithswift.com/articles/161/how-to-use-result-in-swift) to a guide for how that works.  @henrYoda implemented this in his current PR, which you can see [here](https://github.com/pennlabs/penn-mobile-ios/blob/automaticlogin/PennMobile/Auth/PlatformNetworkManager.swift#L28). The idea is we want to pass back the `NetworkingError`, which will contain the reason for failure (parsing error, authentication error, etc.). If it's an authentication error, we want to popup a login controller so they can log back in, and if it's a parsing error, we should throw an alert saying that it was unable to fetch your code.